src/datum_type.cc

```diff
@@ -52,4 +56,24 @@ uint32_t DatumType::SizeInBytes() const {
   return bytes;
 }
 
+std::unique_ptr<Format> DatumType::AsFormat() const {
+  uint32_t bits_per_element = ElementSizeInBytes() * 8;
+  static const char* prefixes = "RGBA";
```

Comment:
We're trusting that row_count_ is at most 4.  Since the data is already validated at parse time, this is ok.
I would normally prefer a bounds assert.